server/src/test/java/io/druid/curator/announcement/AnnouncerTest.java

```diff
@@ -72,38 +73,53 @@ public void testSanity() throws Exception
 
     announcer.start();
 
-    Assert.assertArrayEquals("/test1 has data", billy, curator.getData().decompressed().forPath(testPath1));
-    Assert.assertNull("/somewhere/test2 still does not exist", curator.checkExists().forPath(testPath2));
+    try {
+      Assert.assertArrayEquals("/test1 has data", billy, curator.getData().decompressed().forPath(testPath1));
+      Assert.assertNull("/somewhere/test2 still does not exist", curator.checkExists().forPath(testPath2));
 
-    announcer.announce(testPath2, billy);
+      announcer.announce(testPath2, billy);
 
-    Assert.assertArrayEquals("/test1 still has data", billy, curator.getData().decompressed().forPath(testPath1));
-    Assert.assertArrayEquals("/somewhere/test2 has data", billy, curator.getData().decompressed().forPath(testPath2));
+      Assert.assertArrayEquals("/test1 still has data", billy, curator.getData().decompressed().forPath(testPath1));
+      Assert.assertArrayEquals("/somewhere/test2 has data", billy, curator.getData().decompressed().forPath(testPath2));
 
-    final CountDownLatch latch = new CountDownLatch(1);
-    curator.getCuratorListenable().addListener(
-        new CuratorListener()
-        {
-          @Override
-          public void eventReceived(CuratorFramework client, CuratorEvent event) throws Exception
+      final CountDownLatch latch = new CountDownLatch(1);
+      curator.getCuratorListenable().addListener(
+          new CuratorListener()
           {
-            if (event.getType() == CuratorEventType.CREATE && event.getPath().equals(testPath1)) {
-              latch.countDown();
+            @Override
+            public void eventReceived(CuratorFramework client, CuratorEvent event) throws Exception
+            {
+              if (event.getType() == CuratorEventType.CREATE && event.getPath().equals(testPath1)) {
+                latch.countDown();
+              }
             }
           }
-        }
-    );
-    curator.delete().forPath(testPath1);
-    Assert.assertTrue("Wait for /test1 to be created", timing.forWaiting().awaitLatch(latch));
-
-    Assert.assertArrayEquals("expect /test1 data is restored", billy, curator.getData().decompressed().forPath(testPath1));
-    Assert.assertArrayEquals("expect /somewhere/test2 is still there", billy, curator.getData().decompressed().forPath(testPath2));
+      );
+      curator.inTransaction().delete().forPath(testPath1).and().commit();
```

Comment:
not sure what the semantic difference in curator.delete().forPath(..) vs curator.inTransaction().delete().forPath(..).and().commit() .